### PR TITLE
Code quality csa profiles

### PR DESCRIPTION
**Pull request by jipea**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java [hunk 1]
```diff
@@ -39,16 +39,34 @@ public String getFormat() {
...
+            int maxNbEntries = 200;
+            int maxSizeEntry = 10000000;
+            int countEntries = 0;
+            while ((zipEntry = zipInputStream.getNextEntry()) != null && countEntries < maxNbEntries) {
+                countEntries++;
                 if (!zipEntry.isDirectory()) {
                     FaraoLoggerProvider.BUSINESS_LOGS.info("csa profile crac import : import of file {}", zipEntry.getName());
+                    int currentSizeEntry = 0;
                     File tempFile = File.createTempFile("faraoCsaProfile", ".tmp");
+                    tempFile.setExecutable(true, true);
```
Comment (benrejebmoh) [inline data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java]: i think we can remove this line

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java [hunk 2]
```diff
@@ -39,16 +39,34 @@ public String getFormat() {
...
     public CsaProfileCrac importNativeCrac(InputStream inputStream) {
         TripleStore tripleStoreCsaProfile = TripleStoreFactory.create(CsaProfileConstants.TRIPLESTORE_RDF4J_NAME);
-        ZipInputStream zipInputStream = new ZipInputStream(inputStream);
         ZipEntry zipEntry;
-        try {
-            while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+        try (ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+            //max number of entries and max size of entry are checked to avoid ddos attack with malicious zip file
+            int maxNbEntries = 200;
+            int maxSizeEntry = 10000000;
```
Comment (benrejebmoh) [inline data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java:46]: maybe we can think of something configurable on application side ? i am not sure 10 megabytes are enough for real use cases

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java (hunk 1 above)
Comment (jipea) [reply data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java]: ok, done

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java (hunk 2 above)
Comment (jipea) [reply data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/importer/CsaProfileCracImporter.java:46]: ok, I change it to 10 gigabytes and add a "to do" to parameterize it

**Review by benrejebmoh — APPROVED** [review]

**Review by pet-mit — APPROVED** [review]

**Pull request merged**